src/API/Authentication.php

```diff
@@ -77,6 +82,8 @@ public function __construct(
     /**
      * Set an ApiClient for use in this object
      *
+     * TODO: deprecate
```

Comment:
Any reason why this can't be deprecated now?  If there is it might be a good idea to put when it can actually be deprecated next to the TODO so somebody looking at this in x months knows it was missed and can be done.

Reply:
We do "actual" deprecations in separate PRs so they appear in the CHANGELOG. I'll go through after this PR (or before the release) and put those all through.